#169482450 fix get entry errors and updates tests

Pacifique78 opened the pull request:
#### What does this PR do?
>-This fix add-entries errors
>-Fix eslint errors

#### Description of Task to be completed
>-change name of entryTests file
>-add missing functionalities
>-add another tester
>-fix tests

#### How should this be manually tested?
>- clone the repo
>- run `npm install` and `npm run dev` to start the server
>- use either postman or any other testing tool to see if all endpoints are working 
>- run `npm test` to see if all the test are passing
>- Check if the Travis test are passing

#### What are the relevant pivotal tracker stories?
#169482450

#### Screenshots (if appropriate)
![Screenshot (93)](https://user-images.githubusercontent.com/50268181/68005700-c86a4f80-fc7e-11e9-8a19-202209073fff.png)
![Screenshot (94)](https://user-images.githubusercontent.com/50268181/68005701-c86a4f80-fc7e-11e9-80a9-7b2978f0d8e5.png)
![Screenshot (95)](https://user-images.githubusercontent.com/50268181/68005702-c86a4f80-fc7e-11e9-96a8-a4bfe77f2310.png)
![Screenshot (96)](https://user-images.githubusercontent.com/50268181/68005704-c902e600-fc7e-11e9-915a-6a36f252977f.png)
![Screenshot (97)](https://user-images.githubusercontent.com/50268181/68005708-c902e600-fc7e-11e9-838f-4f2e5a91b840.png)

Pacifique78 committed:
ch(fix errors): fix get entry errors

-change name of entryTests file
-add missing functionalities
-add onether testUser
-fix tests

{Finishes #169482450]

Kasulejoseph commented on Server/Test/v_entryTests.js:
```diff
@@ -115,12 +169,22 @@ describe('Get all entries', () => {
             .set('Authorization', testEntry[8].token)
             .end((err, res) => {
-                expect(res).to.have.status(401);
+                expect(res).to.have.status(400);
```
This should not be a 400, you need to do more research on the status codes. Change to the right status code after wards

Kasulejoseph commented on Server/Test/v_entryTests.js:
```diff
@@ -137,7 +201,7 @@ describe('Get specific entry', () => {
             .set('Authorization', testEntry[8].token)
             .end((err, res) => {
-                expect(res).to.have.status(401);
+                expect(res).to.have.status(400);
```
The status code isn't correct.

Kasulejoseph commented on Server/Test/v_entryTests.js:
```diff
@@ -181,7 +255,7 @@ describe('Delete an entry', () => {
             .set('Authorization', testEntry[8].token)
             .end((err, res) => {
-                expect(res).to.have.status(401);
+                expect(res).to.have.status(400);
```
Same here

Kasulejoseph commented on Server/middleware/checkToken.js:
```diff
@@ -1,17 +1,31 @@
+    const { authorization } = req.headers;
+    if (!authorization) {
+        return res.status(400).json({
+            status: 400,
```
check the status code and use the correct one.

Kasulejoseph commented on Server/middleware/checkToken.js:
```diff
@@ -1,17 +1,31 @@
+        if (!userFound) {
+            return res.status(401).json({
+                status: 401,
+                error: 'You are not authorized to perform this',
```
This `You are not authorized to perform this` message looks incomplete. Complete and should be meaningful enough.

Kasulejoseph commented on Server/middleware/checkToken.js:
```diff
@@ -1,17 +1,31 @@
+        const userFound = await users.find(user => user.id === req.tokenData.id);
+        if (!userFound) {
+            return res.status(401).json({
+                status: 401,
```
The status code is also not the most appropriate one.

Kasulejoseph commented on Server/Model/mockData/testEntry.js:
```diff
@@ -20,6 +20,10 @@ const testEntry = [
     }, {
         token: '',
+    }, {
+        token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MiwiZW1haWwiOiJzZWNvbmRVU2VySWRAZ21haWwuY29tIiwiaWF0IjoxNTcyNDk0MTAxLCJleHAiOjE1NzMwOTg5MDF9.UG_SFBT9mBw7fCMjpMGcNT1L6BVbGpI_09c0AtTZK8U',
```
Hide these tokens, move them to the `.env` file.. It's a security vulnerability to dispose them

Pacifique78 committed:
ch(fix errors): fix error on status code errors

-fix HTTP status codes
-remove check token data from test data to invornoment valiables

{Improves #169482450]

Kasulejoseph reviewed (approved):
Looks good.
